internal/terraform/terraform.go

```diff
@@ -0,0 +1,131 @@
+package terraform
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+)
+
+type Plan struct {
+	ResourceChanges []ResourceChange `json:"resource_changes"`
+	PriorState      PriorState       `json:"prior_state"`
+	Configuration   Configuration    `json:"configuration"`
+}
+
+type PriorState struct {
+	Values PriorStateValues `json:"values"`
+}
+
+type PriorStateValues struct {
+	RootModule PriorStateModule `json:"root_module"`
+}
+
+type PriorStateModule struct {
+	Resources []PriorStateResource `json:"resources"`
+}
+
+type PriorStateResource struct {
+	Type    ResourceType             `json:"type"`
+	Name    string                   `json:"name"`
+	Address string                   `json:"address"`
+	Values  PriorStateResourceValues `json:"values"`
+}
+
+type PriorStateResourceValues struct {
+	InternalUserID   string  `json:"internal_user_id"`
+	ExternalUserID   string  `json:"external_user_id"`
+	Tag              []Tag   `json:"tag"`
```

Comment:
Is Tag property available for all the PriorStateResources regardless of their Type?
I suspect this works for now but in the near future the checker will be handling untaggable resources which can cause problematic situations.

Reply:
If the resource does not have property called "tag" in the json data, the json parsing will simply not include the field in the final struct. And you are right, these should actually all be pointer strings, a.k.a optional since not all resources have tagging capability 👍🏼

Reply:
It could also be a "TaggableStateResourceValue" as an alternative but we are probably fine with this for the time being. More of an heads-up comment.